packages/core/src/components/forms/_controls.scss

```diff
@@ -300,6 +300,10 @@ $control-indicator-spacing: $pt-grid-size !default;
 
   $switch-indicator-text-font-size: 0.7em;
 
+  $switch-text-color: $pt-text-color !default;
```

Comment:
Explicitly setting text colors for switch's inner text (for light/dark and enabled/disabled)